nntrainer/models/neuralnet.cpp

```diff
@@ -1025,4 +1026,13 @@ void NeuralNetwork::print(std::ostream &out, unsigned int flags,
 
   /// @todo Add status to check neuralnet has been run. #290
 }
+
+void NeuralNetwork::forEachLayer(
+  std::function<void(ml::train::Layer &, RunLayerContext &, void *)> fn,
+  void *user_data) {
+  for (auto iter = model_graph.cbegin(); iter != model_graph.cend(); iter++) {
```

Comment:
you can add a quick check that model if model is not initialize, then throw.
or maybe ln->getRunContext() will throw?

Reply:
Yes it throws so I skipped adding it

Reply:
cool.